tendrl/commons/objects/cluster/atoms/import_cluster/__init__.py

```diff
@@ -186,20 +186,38 @@ def run(self):
                         )
                         return False
                     time.sleep(10)
-                    finished = True
+                    child_job_finished_count = 0
+                    child_job_failed = []
                     for child_job_id in parent_job.children:
                         child_job = NS.tendrl.objects.Job(
                             job_id=child_job_id
                         ).load()
-                        if child_job.status != "finished":
-                            finished = False
-                            break
-                    if finished:
+                        if child_job.status == "finished":
+                            child_job_finished_count += 1
+                        if child_job.status == "failed":
+                            child_job_finished_count += 1
```

Comment:
child_job_finished_count += 1 ??

Reply:
@shtripat i am adding total number of child_jobs completed, which means even job is failed that is also i am considering as completed.

So if all child are completed then i am comparing with the total number of child job is equal to completed then i am stoping parent job wait

Reply:
i will mofdify child job child_job_finished_count to child_job_completed_count

Reply:
problem is if any child job is failed then parent is wating for timeout.  using this logic, it will stop waiting imediatly when all jobs are completed with finshed or failed

Reply:
Ack